src/git.rs

```diff
@@ -83,46 +83,84 @@ impl Parser {
     fn sections(&self, git_diff: &str) -> Vec<Section> {
         let mut sections = Vec::new();
         let mut file_name = "";
+        let mut line_start = 0;
+        let mut line_end = 0;
+        let mut current_line = 0;
+
         for l in git_diff.lines() {
-            // Add or edit a file
+            // File added or edited
             // +++ b/Cargo.lock
             if l.starts_with("+++") {
                 // TODO: do something less ugly with the bounds and indexing
                 file_name = l[l.find('/').unwrap() + 1..].into();
             }
+
             // Actual diff lines
-            // @@ -33,6 +33,9 @@ version = "0.1.0"
+            // '@@ -33,6 +33,9 @@ version = "0.1.0"'
             if l.starts_with("@@") {
-                // For now, we will focus on the added lines.
-                // @@ and space
-                let after_ats = &l[3..];
-                // space and @@
-                let before_second_ats_index = &after_ats.find("@@").unwrap() - 1;
-                let diff_lines = &after_ats[..before_second_ats_index];
-                // -33,6 +33,9
-                let (_, a) = diff_lines.split_at(diff_lines.find(' ').unwrap());
-                let added = a.trim();
-
-                let (added_start, added_span) = if let Some(index) = added[1..].find(',') {
-                    let (a, b) = added[1..].split_at(index);
-                    (a, &b[1..])
+                current_line = get_diff_line_start(&l) - 1;
+            }
+
+            // Increase the current line counter if added or untouched line in diff
+            if l.starts_with(' ') || (l.starts_with('+') && !l.starts_with("+++")) {
```

Comment:
This checks if a line is an edit or an add ?

Reply:
Yes, only on edited and added lines (since deleted lines do not account (obviously) for the line number.
This allows us to keep track of the current line number of the line 'l' we are iterating over.